mlrun/utils/vault.py

```diff
@@ -0,0 +1,269 @@
+
+        if not response:
+            return secrets
+
+        values = response.json()["data"]["data"]
+
+        # if no specific keys were asked for, return all the values available
+        if not keys:
+            return values
+
+        for key in keys:
+            if key in values:
+                secrets[key] = values[key]
+        return secrets
+
+    def add_vault_secrets(self, items, project=None, user=None):
+        data_object = {"data": items}
+        url = VaultStore._generate_vault_path(project=project, user=user)
+
+        response = self._vault_api_call("POST", url, data_object)
+        if not response:
+            raise ValueError(
+                f"Vault failed the API call to create secrets. project={project}/user={user}"
+            )
+
+    def delete_vault_secrets(self, project=None, user=None):
+        self._login()
+        # Using the API to delete all versions + metadata of the given secret.
+        url = "v1/secret/metadata/" + VaultStore._generate_vault_path(
+            prefix="", project=project, user=user
+        )
+
+        response = self._vault_api_call("DELETE", url)
+        if not response:
+            raise ValueError(
+                f"Vault failed the API call to delete secrets. project={project}/user={user}"
+            )
+
+    def create_project_policy(self, project):
+        policy_name = "mlrun-project-{}".format(project)
+        # TODO - need to make sure name is escaped properly and invalid chars are stripped
+        url = "v1/sys/policies/acl/" + policy_name
+
+        policy_str = (
+            'path "secret/data/mlrun/projects/{0}" {{\n'
+            + '  capabilities = ["read", "list", "create", "delete", "update"]\n'
+            + "}}\n"
+            + 'path "secret/data/mlrun/projects/{0}/*" {{\n'
+            + '  capabilities = ["read", "list", "create", "delete", "update"]\n'
+            + "}}"
+        ).format(project)
+
+        data_object = {"policy": policy_str}
+
+        response = self._vault_api_call("PUT", url, data_object)
+        if not response:
+            raise ValueError(
+                "Vault failed the API call to create a policy. Response code: ({}) - {}".format(
+                    response.status_code, response.reason
+                )
+            )
+        return policy_name
+
+    def create_project_role(self, project, sa, policy, namespace="default-tenant"):
+        role_name = "mlrun-role-project-{}".format(project)
+        # TODO - need to make sure name is escaped properly and invalid chars are stripped
+        url = "v1/auth/kubernetes/role/" + role_name
+
+        role_object = {
+            "bound_service_account_names": sa,
+            "bound_service_account_namespaces": namespace,
+            "policies": [policy],
+            "token_ttl": 1800000,
+        }
+
+        response = self._vault_api_call("POST", url, role_object)
+        if not response:
+            raise ValueError(
```

Comment:
General comment, use `mlrun.errors.MLRunInvalidArgumentError` instead of `ValueError`